tool/tctl/common/db_command.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2020 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"context"
+	"os"
+
+	"github.com/gravitational/teleport"
+	"github.com/gravitational/teleport/lib/auth"
+	"github.com/gravitational/teleport/lib/defaults"
+	"github.com/gravitational/teleport/lib/service"
+	"github.com/gravitational/teleport/lib/services"
+
+	"github.com/gravitational/kingpin"
+	"github.com/gravitational/trace"
+)
+
+// DbCommand implements "tctl db" group of commands.
+type DbCommand struct {
+	config *service.Config
+
+	// format is the output format (text, json, or yaml).
+	format string
+
+	// dbList implements the "tctl db ls" subcommand.
+	dbList *kingpin.CmdClause
+}
+
+// Initialize allows DBCommand to plug itself into the CLI parser.
+func (c *DbCommand) Initialize(app *kingpin.Application, config *service.Config) {
+	c.config = config
+
+	db := app.Command("db", "Operate on databases registered with the cluster.")
+	c.dbList = db.Command("ls", "List all databases registered with the cluster.")
+	c.dbList.Flag("format", "Output format, 'text', 'json', or 'yaml'").Default("text").StringVar(&c.format)
+}
+
+// TryRun attempts to run subcommands like "apps ls".
+func (c *DbCommand) TryRun(cmd string, client auth.ClientI) (match bool, err error) {
+	switch cmd {
+	case c.dbList.FullCommand():
+		err = c.ListDatabases(client)
+	default:
+		return false, nil
+	}
+	return true, trace.Wrap(err)
+}
+
+// ListDatabases prints the list of database proxies that have recently sent
+// heartbeats to the cluster.
+func (c *DbCommand) ListDatabases(client auth.ClientI) error {
+	servers, err := client.GetDatabaseServers(context.TODO(), defaults.Namespace, services.SkipValidation())
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	coll := &dbCollection{servers: servers}
+	switch c.format {
+	case teleport.Text:
+		err = coll.writeText(os.Stdout)
+	case teleport.JSON:
+		err = coll.writeJSON(os.Stdout)
+	case teleport.YAML:
+		err = coll.writeYAML(os.Stdout)
+	default:
+		return trace.BadParameter("unknown format %q", c.format)
+	}
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	return nil
+}
+
+const dbMessage = `The invite token: %v.
```

Comment:
Have you considered using a text template for this? This looks like a better fit than the `fmt` package.